Consider changing npm 'prepack' to 'prepare'

<!-- Not filling out ALL of the relevant fields in this issue will cause your issue to be closed -->

### Summary:
Right now Dynamoose is doing TypeScript compilation inside npm 'prepack' lifecycle script, which runs right before the files are packaged and deployed to the registry. The problem is that if dev dependencies of the package are not installed before 'prepack' is run it will fail. For that reason installing Dynamoose from git fails, as tsc is run without installing dev dependencies. A more commonly used script for this job is 'prepare', which works almost the same way, but forces dev dependencies to be installed, and so does not have this problem.



### Code sample:
```shell
$ npm install git+ssh://git@github.com/dynamoosejs/dynamoose.git
```

### Current output and behavior:
```shell
> dynamoose@1.9.0 prepack /home/dols3m/.npm/_cacache/tmp/git-clone-7923d6a4
> tsc

lib/Dynamoose.ts:1:19 - error TS2307: Cannot find module 'https'.

1 import https from 'https';
                    ~~~~~~~

lib/Dynamoose.ts:2:17 - error TS2307: Cannot find module 'aws-sdk'.

2 import AWS from 'aws-sdk';
                  ~~~~~~~~~

lib/Dynamoose.ts:3:23 - error TS2307: Cannot find module 'debug'.

3 import debugBase from 'debug';
                        ~~~~~~~

lib/Dynamoose.ts:4:15 - error TS2307: Cannot find module 'q'.

4 import Q from 'q';
                ~~~


Found 4 errors.

npm ERR! premature close
```



### Expected output and behavior:
```shell
+ dynamoose@1.9.0
```



### Environment:

Operating System: Linux
Operating System Version: 4.18.0-24-generic
Node.js version (`node -v`): v10.13.0
NPM version: (`npm -v`): 6.5.0
Dynamoose version: 1.9.0
Dynamoose Plugins:
<!-- Please include the name, GitHub & NPM links to ALL of the plugins you are using with Dynamoose -->


### Dynamoose Plugins:

- [ ] Yes, I believe that one or multiple 3rd party Dynamoose plugins are effecting this issue
- [ ] No, I believe this issue is independent of any 3rd party Dynamoose plugins I'm using
- [ ] Unknown, I'm unsure if Dynamoose plugins are effecting this issue
- [x] I am not using any Dynamoose plugins


### Type (select 1):
- [ ] Bug report
- [ ] Feature suggestion
- [ ] Question
- [x] Other suggestion
- [ ] Something not listed here


### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have ensured that all of my plugins that I'm using with Dynamoose are listed above
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
     "lint": "eslint .",
     "tslint": "tslint lib/**/*.ts",
     "coverage": "nyc npm test",
-    "prepack": "tsc",
+    "prepare": "tsc",
     "commit": "git-cz",
     "sem-rel": "semantic-release",
     "lint:commit": "commitlint -E $GIT_PARAMS",

